data loss: auto upload does not notice some files

### Steps to reproduce
1. Use auto upload to upload files to nextcloud
2. Check regularly whether it uploads all files

### Expected behaviour
- it should never fail to notice a new file. Maybe fail uploading but not "failing silently", i.e. just ignore some new file in the folder set to upload to nexrcloud

### Actual behaviour
- Sometimes the app ignores a file without any error message, i.e. it uploads files before that without issues and also after that but never notices that it lacks one existing file. The file is smaller than others that are uploaded fine. It never notices it, not even when it starts to upload more pictures the next day.

This will cause data loss since users rely on the app catching all new files for upload! I just hope I noticed all and was lucky! If users check the uploads section of the app and there are no error messages they must rely on the app that all files before the ones uploaded last have already been uploaded. It is not feasible to have to double-check for each file whether it was really uploaded.

Restarting the smartphone does not help.

### Can you reproduce this problem on https://try.nextcloud.com?
I currently have one file which is shown as thumbnail in the auto upload settings, which shows 8 previews for each folder. If somebody can tell me how to debug, I could reproduce that it is not found when refreshing the uploads section of the app.

### Environment data
Android version:
8.1
Device model: 
Nexus 5x
Stock or customized system:
Lineage os
Nextcloud app version:
3.11.1
Nextcloud server version:
18.0.4
Reverse proxy:
None
### Logs
#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

Yes. It's quite a file. I'm not really sure what I'm looking for but I found this near the end of the file where it seems that a sync is being attempted. I hope this helps.

`
10-26 21:31:36.866 24718 24739 E WM-WorkerWrapper: Work [ id=bc75bddd-e643-4cbd-8365-69d9f3ca3780, tags={ name:periodic_files_sync, com.nextcloud.client.jobs.FilesSyncWork, *, timestamp:1603714595529 } ] failed because it threw an exception/error
10-26 21:31:36.866 24718 24739 E WM-WorkerWrapper: java.util.concurrent.ExecutionException: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.io.File.getAbsolutePath()' on a null object reference
10-26 21:31:36.866 24718 24739 E WM-WorkerWrapper: 	at androidx.work.impl.utils.futures.AbstractFuture.getDoneValue(AbstractFuture.java:516)
10-26 21:31:36.866 24718 24739 E WM-WorkerWrapper: 	at androidx.work.impl.utils.futures.AbstractFuture.get(AbstractFuture.java:475)
10-26 21:31:36.866 24718 24739 E WM-WorkerWrapper: 	at androidx.work.impl.WorkerWrapper$2.run(WorkerWrapper.java:298)
10-26 21:31:36.866 24718 24739 E WM-WorkerWrapper: 	at androidx.work.impl.utils.SerialExecutor$Task.run(SerialExecutor.java:91)
10-26 21:31:36.866 24718 24739 E WM-WorkerWrapper: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
10-26 21:31:36.866 24718 24739 E WM-WorkerWrapper: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
10-26 21:31:36.866 24718 24739 E WM-WorkerWrapper: 	at java.lang.Thread.run(Thread.java:919)
10-26 21:31:36.866 24718 24739 E WM-WorkerWrapper: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.io.File.getAbsolutePath()' on a null object reference
10-26 21:31:36.866 24718 24739 E WM-WorkerWrapper: 	at com.owncloud.android.utils.FileStorageUtils.getInstantUploadFilePath(FileStorageUtils.java:181)
10-26 21:31:36.866 24718 24739 E WM-WorkerWrapper: 	at com.nextcloud.client.jobs.FilesSyncWork.syncFolder(FilesSyncWork.kt:190)
10-26 21:31:36.866 24718 24739 E WM-WorkerWrapper: 	at com.nextcloud.client.jobs.FilesSyncWork.doWork(FilesSyncWork.kt:119)
10-26 21:31:36.866 24718 24739 E WM-WorkerWrapper: 	at androidx.work.Worker$1.run(Worker.java:85)
10-26 21:31:36.866 24718 24739 E WM-WorkerWrapper: 	... 3 more
10-26 21:31:36.874 24718 24739 I WM-WorkerWrapper: Worker result FAILURE for Work [ id=bc75bddd-e643-4cbd-8365-69d9f3ca3780, tags={ name:periodic_files_sync, com.nextcloud.client.jobs.FilesSyncWork, *, timestamp:1603714595529 } ]
10-26 21:31:37.331 24718 25854 D UploadsStorageManager: QUERY: (status== ? OR last_result==9 OR last_result==13 OR last_result==11 OR last_result==14 ) AND last_result!= 16 ROWID: -1
10-26 21:31:37.343   590   625 D audio_hw_primary: out_standby: enter: stream (0xf4823000) usecase(1: low-latency-playback)
10-26 21:31:37.344 24718 25854 V UploadsStorageManager: getUploads() got 0 rows from page 0, 0 rows total so far, last ID -1
10-26 21:31:37.344 24718 25854 V UploadsStorageManager: getUploads() returning 0 (0) rows after reading 1 pages
10-26 21:31:37.429   590   625 D audio_hw_primary: disable_audio_route: reset and update mixer path: low-latency-playback
10-26 21:31:37.433   590   625 D soundtrigger: audio_extn_sound_trigger_update_stream_status: uc_info->id 1 of type 0 for Event 2, with Raise=0
10-26 21:31:37.433   590   625 D msm8974_platform: platform_split_snd_device: snd_device(2) num devices(0) new_snd_devices(0)
10-26 21:31:37.434   590   625 I msm8974_platform: platform_get_custom_mtmx_params: no matching param with id 0 ip_ch 2 op_ch 2 uc_id 1 snd_dev 2
10-26 21:31:37.434   590   625 D hardware_info: hw_info_append_hw_type : device_name = speaker
10-26 21:31:37.434   590   625 D audio_hw_primary: disable_snd_device: snd_device(2: speaker)
10-26 21:31:37.451   590   625 D hardware_info: hw_info_append_hw_type : device_name = vi-feedback
10-26 21:31:37.451   590   625 D audio_hw_primary: disable_snd_device: snd_device(127: vi-feedback)
10-26 21:31:37.451   590   625 D msm8974_platform: platform_split_snd_device: snd_device(127) num devices(0) new_snd_devices(0)
10-26 21:31:37.451   590   625 I soundtrigger: audio_extn_sound_trigger_update_device_status: device 0x7f of type 1 for Event 0, with Raise=0
10-26 21:31:37.451   590   625 D audio_hw_primary: disable_audio_route: reset and update mixer path: spkr-vi-record
10-26 21:31:37.452   590   625 E audio_hw_extn: audio_extn_set_custom_mtmx_params: invalid input stream for capture usecase id:48
10-26 21:31:37.478   590   625 E msm8974_platform: platform_set_swap_mixer: Could not get ctl for mixer cmd - Swap channel
10-26 21:31:37.479   590   625 I soundtrigger: audio_extn_sound_trigger_update_device_status: device 0x2 of type 0 for Event 0, with Raise=0
10-26 21:31:37.479   590   625 D audio_hw_primary: out_standby: exit
10-26 21:31:39.104  2198  4068 D PowerUI : can't show warning due to - plugged: true status unknown: false
10-26 21:31:42.683 24718 24795 D MediaProvider: Reading images for browserhome
10-26 21:31:42.749 24718 24795 D MediaProvider: Reading images for Pictures
10-26 21:31:42.769 24718 24795 D MediaProvider: Reading images for Conversations Images
10-26 21:31:42.817 24718 24795 D MediaProvider: Reading images for Cambackup
10-26 21:31:42.855 24718 24795 D MediaProvider: Reading images for Camera
10-26 21:31:42.921 24718 24795 D MediaProvider: Reading images for Viber
10-26 21:31:42.975 24718 24795 D MediaProvider: Reading images for Screenshots
10-26 21:31:43.037 24718 24795 D MediaProvider: Reading images for random keepers
10-26 21:31:43.094 24718 24795 D MediaProvider: Reading images for Jordan B. Peterson - 12 Rules for Life An Antidote to Chaos (Unabridged)
10-26 21:31:43.190 24718 24795 D MediaProvider: Reading videos for Cambackup
10-26 21:31:43.270 24718 24795 D MediaProvider: Reading videos for Camera
10-26 21:31:43.344 24718 24795 D MediaProvider: Reading videos for Conversations Videos
10-26 21:31:43.411 24718 24795 D MediaProvider: Reading videos for Viber
10-26 21:31:43.470 24718 24795 V ArbitraryDataProvider: Updating arbitrary data with cloud id: global key: media_folders value: {"imageMediaFolders":["/storage/emulated/0/Documents/browserhome","/storage/emulated/0/Conversations/Media/Conversations Images","/storage/071B-F82B/Cambackup","/storage/emulated/0/DCIM/Camera","/storage/emulated/0/Pictures/Viber","/storage/emulated/0/Pictures/Screenshots","/storage/071B-F82B/random keepers","/storage/emulated/0/Music/Audiobooks/Jordan B. Peterson - 12 Rules for Life An Antidote to Chaos (Unabridged)","/storage/071B-F82B/Cambackup","/storage/emulated/0/DCIM/Camera","/storage/emulated/0/Conversations/Media/Conversations Videos","/storage/emulated/0/Movies/Viber","/storage/emulated/0/Android/media/com.google.android.markup/Markup","/storage/emulated/0/Download","/storage/071B-F82B","/storage/emulated/0/Download/Frost"],"videoMediaFolders":[]}
10-26 21:31:43.508 24718 24741 I WM-WorkerWrapper: Worker result SUCCESS for Work [ id=4d37ce53-e114-4dc1-870d-cb7c0b2be962, tags={ com.nextcloud.client.jobs.MediaFoldersDetectionWork, timestamp:1601589005753, *, name:periodic_media_folder_detection } ]
10-26 21:31:44.328  1008 10082 D ConnectivityService: requestNetwork for uid/pid:10273/24718 NetworkRequest [ TRACK_DEFAULT id=583, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10273] ]
10-26 21:31:44.333  1008  2044 D WifiNetworkFactory: got request NetworkRequest [ TRACK_DEFAULT id=583, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10273] ] with score 60 and serial -1
10-26 21:31:44.334  1008  2129 D Ethernet: got request NetworkRequest [ TRACK_DEFAULT id=583, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10273] ] with score 60 and serial -1
10-26 21:31:44.336  1008  2044 D UntrustedWifiNetworkFactory: got request NetworkRequest [ TRACK_DEFAULT id=583, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10273] ] with score 60 and serial -1
10-26 21:31:44.337  2500  2500 D PhoneSwitcherNetworkRequstListener: got request NetworkRequest [ TRACK_DEFAULT id=583, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10273] ] with score 60 and serial -1
10-26 21:31:44.369   539 25856 E ResolverController: No valid NAT64 prefix (101, <
`

--- a/src/main/java/com/owncloud/android/utils/FileStorageUtils.java
+++ b/src/main/java/com/owncloud/android/utils/FileStorageUtils.java
@@ -175,8 +175,14 @@ public final class FileStorageUtils {
             subfolderByDatePath = getSubPathFromDate(dateTaken, current);
         }
 
-        String relativeSubfolderPath = new File(file.getAbsolutePath().replace(syncedFolderLocalPath, ""))
-            .getParentFile().getAbsolutePath();
+        File parentFile = new File(file.getAbsolutePath().replace(syncedFolderLocalPath, "")).getParentFile();
+
+        String relativeSubfolderPath = "";
+        if (parentFile == null) {
+            Log_OC.e("AutoUpload", "Parent folder does not exists!");
+        } else {
+            relativeSubfolderPath = parentFile.getAbsolutePath();
+        }
 
         // Path must be normalized; otherwise the next RefreshFolderOperation has a mismatch and deletes the local file.
         return (remotePath +

